utilities/flash_network_bootloader.py

```
@@ -0,0 +1,270 @@
+    )
+    args = parser.parse_args()
+
+    if (args.bootloader is None) != (args.expectedVersion is None):
+        parser.error("--bootloader and --expected-version must be provided together")
+    if args.bootloader is not None and not args.bootloader.is_file():
+        parser.error(f"bootloader image does not exist or is not a file: {args.bootloader}")
+
+    return args
+
+
+def versionString(version: Any) -> str:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `version`

(ANN401)